### ISSUE TITLE
How are the AUROC metrics for TF, DHS, and HM calculated in Chromatin Profile prediction?

### ISSUE THREAD

**Issue by ychuest** [HazyResearch/hyena-dna#90]
The 919 chromatin features in Chromatin Profile prediction include 690 TF binding profiles for 160 distinct transcription factors, 125 DHS profiles, and 104 HM profiles. I would like to understand how the AUROC metrics for TF, DHS, and HM are computed during actual processing. In the source code implementation of class ChromatinProfile(HG38), all 919 labels are input simultaneously. This approach does not seem to yield the AUROC metrics for TF, DHS, and HM. How are these metrics calculated within the framework?

class ChromatinProfile(HG38):
    _name_= 'chromatin_profile'
    l_output = 0  # need to set this for decoder to work correctly for seq level
    def __init__(self, data_path, ref_genome_path, ref_genome_version=None,
                 tokenizer_name=None, dataset_config_name=None, 
                 max_length=1000, d_output=2, rc_aug=False, add_eos=True, val_only=False,
                 batch_size=32, batch_size_eval=None, num_workers=1,
                 shuffle=False, pin_memory=False, drop_last=False, fault_tolerant=False, ddp=False,
                 fast_forward_epochs=None, fast_forward_batches=None,
                 *args, **kwargs):
        
        self.data_path = data_path
        self.ref_genome_path = ref_genome_path
        self.ref_genome_version = ref_genome_version
        self.dataset_config_name = dataset_config_name
        self.tokenizer_name = tokenizer_name
        self.d_output = d_output
        self.rc_aug = rc_aug  # reverse compliment augmentation
        self.max_length = max_length
        self.add_eos = add_eos
        
        self.val_only=val_only
        
        self.batch_size = batch_size
        self.batch_size_eval = batch_size_eval if batch_size_eval is not None else self.batch_size
        self.num_workers = num_workers
        self.shuffle = shuffle
        self.pin_memory = pin_memory
        self.drop_last = drop_last

        if fault_tolerant:
            assert self.shuffle
        self.fault_tolerant = fault_tolerant
        if ddp:
            assert fault_tolerant
        self.ddp = ddp
        self.fast_forward_epochs = fast_forward_epochs
        self.fast_forward_batches = fast_forward_batches
        if self.fast_forward_epochs is not None or self.fast_forward_batches is not None:
            assert ddp and fault_tolerant

    def setup(self, stage=None):
        if self.tokenizer_name == 'char':
            print("**Using Char-level tokenizer**")
            self.tokenizer = CharacterTokenizer(
                characters=['A', 'C', 'G', 'T', 'N'],
                model_max_length=self.max_length + 2,  # add 2 since default adds eos/eos tokens, crop later
                add_special_tokens=False,
            )
        elif self.tokenizer_name == 'bpe':
            print("**using pretrained AIRI tokenizer**")
            self.tokenizer = AutoTokenizer.from_pretrained('AIRI-Institute/gena-lm-bert-base')

        self.vocab_size = len(self.tokenizer)
        
        # Create all splits: torch datasets
        if self.val_only:
            splits=['val']*3
        else:
            splits=['train','val','test']
        self.dataset_train, self.dataset_val, self.dataset_test = [
            ChromatinProfileDataset(
                max_length=self.max_length,
                ref_genome_path = self.ref_genome_path,
                ref_genome_version = self.ref_genome_version,
                coords_target_path = f'{self.data_path}/{split}_{self.ref_genome_version}_coords_targets.csv',
                tokenizer=self.tokenizer,
                tokenizer_name=self.tokenizer_name,
                use_padding=True,
            )
            for split in splits
        ]

**Comment by cbirchsy** [comment]
Hi, 

AUROCs are computed for each of the 919 labels individually and averaged within the TF, DHS and HM sets. This is how it was done in the original paper that introduced the task https://pmc.ncbi.nlm.nih.gov/articles/PMC4768299/ 

Hope this helps

**Comment by ychuest** [comment]
> Hi,
> 
> AUROCs are computed for each of the 919 labels individually and averaged within the TF, DHS and HM sets. This is how it was done in the original paper that introduced the task https://pmc.ncbi.nlm.nih.gov/articles/PMC4768299/
> 
> Hope this helps
I redesigned the task code to execute TF, DHS, and HM's AUROC designs separately within the pipeline. Wandb can monitor the real-time AUROC status. Thank you for answering my question.

<img width="2466" height="906" alt="Image" src="https://github.com/user-attachments/assets/a6eba8a2-f027-4b78-a424-df3b48bb0705" />
